jest.config.js

```diff
@@ -3,6 +3,7 @@ module.exports = {
     'src/components/**/*.js?(x)',
     '!src/**/*.story.js?(x)',
     '!src/**/hooks/*.js',
+    '!src/**/util/*.js',
```

Comment:
Is there a reason that this shouldn't be tested?

Reply:
I agree this should probably be covered in tests if others are relying on it. All the API calls would need to be mocked though. @blechner what were your thoughts on exempting this? At the very least this glob should be much more specific, we don't want to ignore all files in `util` folders.
```suggestion
    '! src/components/SuiteHeader/util/suiteHeaderData.js',
```

Reply:
Considering this is the header that multiple products will be using, I'd think we would want even more coverage than our usual threshold, especially with data fetching involved

Reply:
How about we open an issue for increasing util/ test coverage and prioritize it accordingly?  This specific PR doesn't contain any logic changes (just refactoring), so it would make sense to me to contribute tests in a separate logical unit of work.

Reply:
This PR is just a refactor, test coverage could be added later but should be a separate piece of work. I'd welcome some help mocking out the API layer, i'm happy to come along and add the logical tests once that is done.

Reply:
btw @tay1orjones I've added your suggested commit, but it had a typo, a space between the `!` and the `src/`, it is fixed now 👍

Reply:
@blechner thanks! Sorry for the typo. 

Non-experimental components in the storybook are required to have 80% test coverage. That said, we haven't defined or documented when/how best to test hooks #1615. I have opened https://github.com/carbon-design-system/carbon-addons-iot-react/issues/1810 to specifically track test coverage for SuiteHeader and it's subcomponents/hooks.